### PR TITLE
Updating google.golang.org/api version to v0.45.0

### PR DESCRIPTION
**Pull request by basantsa1989**
Updating to latest google.golang.org/api (version v0.45.0)

/kind design
/kind api-change

```release-note
NONE
```

Commands run:

##Commit ID points to the commit tagging the v0.45.0 release
hack/pin-dependency.sh google.golang.org/api de9f25cba16a59e6a62c6b97478882f5392143f5
hack/update-vendor.sh
hack/verify-vendor.sh

Subsequently fixed transitive dependencies by pining, update and verify sequences.

List and sequence of dependencies pinned:

```
hack/pin-dependency.sh google.golang.org/api de9f25cba16a59e6a62c6b97478882f5392143f5
hack/update-vendor.sh
hack/verify-vendor.sh

hack/pin-dependency.sh cloud.google.com/go v0.81.0
hack/pin-dependency.sh github.com/go-openapi/jsonpointer v0.19.5
hack/pin-dependency.sh github.com/go-openapi/jsonreference v0.19.5
hack/pin-dependency.sh github.com/go-openapi/swag v0.19.14
hack/pin-dependency.sh go.opencensus.io v0.23.0
hack/pin-dependency.sh golang.org/x/oauth2 v0.0.0-20210413134643-5e61552d6c78
hack/pin-dependency.sh google.golang.org/appengine v1.6.7
hack/verify-vendor.sh

hack/pin-dependency.sh cloud.google.com/go/storage v1.10.0
hack/pin-dependency.sh github.com/golang/mock v1.5.0
hack/pin-dependency.sh github.com/google/pprof v0.0.0-20210226084205-cbba55b83ad5
hack/pin-dependency.sh github.com/mailru/easyjson v0.7.6
hack/verify-vendor.sh

hack/pin-dependency.sh cloud.google.com/go/bigquery v1.8.0
hack/pin-dependency.sh github.com/ianlancetaylor/demangle v0.0.0-20200824232613-28f6c0f3b639
hack/verify-vendor.sh
hack/update-vendor.sh

hack/pin-dependency.sh honnef.co/go/tools v0.0.1-2020.1.4
hack/update-vendor.sh
hack/verify-vendor.sh
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -41,11 +41,12 @@ require (
 	github.com/evanphx/json-patch v4.11.0+incompatible
 	github.com/fsnotify/fsnotify v1.4.9
 	github.com/go-bindata/go-bindata v3.1.1+incompatible
+	github.com/go-openapi/spec v0.20.3
```
Comment (liggitt) [inline go.mod:44]: what is adding this back in? we've worked hard to drop it, and are in the process of adding a presubmit to keep it from being reintroduced

Comment (basantsa1989) [reply go.mod:44]: Added the full sequence in the first comment/ PR description.

**Comment by liggitt** [comment]
verify-dependencies is failing, showing that file is not needed:

```
@@ -41,7 +41,6 @@ require (
 	github.com/evanphx/json-patch v4.11.0+incompatible
 	github.com/fsnotify/fsnotify v1.4.9
 	github.com/go-bindata/go-bindata v3.1.1+incompatible
-	github.com/go-openapi/spec v0.20.3
 	github.com/go-ozzo/ozzo-validation v3.5.0+incompatible // indirect
```

try running `make clean generated_files` to make sure you don't have old generated files still referencing that

**Comment by basantsa1989** [comment]
> verify-dependencies is failing, showing that file is not needed:
> 
> ```
> @@ -41,7 +41,6 @@ require (
>  	github.com/evanphx/json-patch v4.11.0+incompatible
>  	github.com/fsnotify/fsnotify v1.4.9
>  	github.com/go-bindata/go-bindata v3.1.1+incompatible
> -	github.com/go-openapi/spec v0.20.3
>  	github.com/go-ozzo/ozzo-validation v3.5.0+incompatible // indirect
> ```
> 
> try running `make clean generated_files` to make sure you don't have old generated files still referencing that

Removed github.com/go-openapi/spec update after "make clean generated_files"
hack/update-vendor.sh and hack/verify-vendor.sh did not complain after this. Hoping this resolves it.

File: go.mod [hunk 2]
```diff
@@ -255,9 +254,10 @@ replace (
 	github.com/go-kit/log => github.com/go-kit/log v0.1.0
 	github.com/go-logfmt/logfmt => github.com/go-logfmt/logfmt v0.5.0
 	github.com/go-logr/logr => github.com/go-logr/logr v0.4.0
-	github.com/go-openapi/jsonpointer => github.com/go-openapi/jsonpointer v0.19.3
-	github.com/go-openapi/jsonreference => github.com/go-openapi/jsonreference v0.19.3
-	github.com/go-openapi/swag => github.com/go-openapi/swag v0.19.5
+	github.com/go-openapi/jsonpointer => github.com/go-openapi/jsonpointer v0.19.5
+	github.com/go-openapi/jsonreference => github.com/go-openapi/jsonreference v0.19.5
+	github.com/go-openapi/spec => github.com/go-openapi/spec v0.20.3
```
Comment (liggitt) [inline go.mod:259]: I expect `hack/lint-dependencies.sh` would show this should be dropped as well

Comment (basantsa1989) [reply go.mod]: Ran hack/lint-dependencies.sh and handled the following:

```
$ hack/lint-dependencies.sh

Use the given commands to remove pinned module versions that aren't actually used:
GO111MODULE=on go mod edit -dropreplace cloud.google.com/go/datastore
GO111MODULE=on go mod edit -dropreplace cloud.google.com/go/pubsub
GO111MODULE=on go mod edit -dropreplace github.com/google/martian
GO111MODULE=on go mod edit -dropreplace github.com/go-openapi/spec
GO111MODULE=on go mod edit -dropreplace rsc.io/quote/v3
GO111MODULE=on go mod edit -dropreplace rsc.io/sampler
All pinned versions of checked dependencies match their preferred version.
```

**Comment by pacoxu** [comment]
After running `hack/pin-dependency.sh xxx xxx`, run  `hack/lint-dependencies.sh`  and follow its suggestions. 
Util `hack/lint-dependencies.sh` return no errors,  just run `hack/update-vendor.sh`  once.

It would save some time.

> BTW, a small improvement for `hack/pin-dependency.sh` would be triggering `hack/lint-dependencies.sh` after it.

**Comment by basantsa1989** [comment]
@liggitt 
To resolve "pull-kubernetes-dependencies" test, I wan hack/update-vendor-licenses.sh. I have a new file generated: LICENSES/vendor/github.com/josharian/intern/LICENSE
Should I be adding this LICENSE file.

**Comment by basantsa1989** [comment]
> After running `hack/pin-dependency.sh xxx xxx`, run `hack/lint-dependencies.sh` and follow its suggestions.
> Util `hack/lint-dependencies.sh` return no errors, just run `hack/update-vendor.sh` once.
> 
> It would save some time.
> 
> > BTW, a small improvement for `hack/pin-dependency.sh` would be triggering `hack/lint-dependencies.sh` after it. Currently, it suggest to run `hack/update-vendor.sh` directly.

Ack. Thanks @pacoxu

**Comment by dims** [comment]
@basantsa1989 one more thing to fix!

```
--- vendor/k8s.io/controller-manager/go.sum	2021-06-17 22:16:58.546777261 +0000
+++ /home/prow/go/src/k8s.io/kubernetes/_tmp/kube-vendor.O0NPIq/kubernetes/vendor/k8s.io/controller-manager/go.sum	2021-06-17 22:19:45.686223648 +0000
@@ -714,7 +714,6 @@ golang.org/x/oauth2 v0.0.0-2018082121233
 golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421/go.mod h1:gOpvHmFTYa4IltrdGE7lF6nIHvwfUNPOp7c8zoXwtLw=
 golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45/go.mod h1:gOpvHmFTYa4IltrdGE7lF6nIHvwfUNPOp7c8zoXwtLw=
 golang.org/x/oauth2 v0.0.0-20191202225959-858c2ad4c8b6/go.mod h1:gOpvHmFTYa4IltrdGE7lF6nIHvwfUNPOp7c8zoXwtLw=
-golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d h1:TzXSXBo42m9gQenoE3b9BGiEpg5IG2JkU5FkPIawgtw=
 golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d/go.mod h1:gOpvHmFTYa4IltrdGE7lF6nIHvwfUNPOp7c8zoXwtLw=
 golang.org/x/oauth2 v0.0.0-20200902213428-5d25da1a8d43/go.mod h1:KelEdhl1UZF7XfJ4dDtk6s++YSgaE7mD/BuKKDLBl4A=
 golang.org/x/oauth2 v0.0.0-20201109201403-9fd604954f58/go.mod h1:KelEdhl1UZF7XfJ4dDtk6s++YSgaE7mD/BuKKDLBl4A=
Vendor Verify failed.
```

**Comment by basantsa1989** [comment]
> @basantsa1989 one more thing to fix!
> 
> ```
> --- vendor/k8s.io/controller-manager/go.sum	2021-06-17 22:16:58.546777261 +0000
> +++ /home/prow/go/src/k8s.io/kubernetes/_tmp/kube-vendor.O0NPIq/kubernetes/vendor/k8s.io/controller-manager/go.sum	2021-06-17 22:19:45.686223648 +0000
> @@ -714,7 +714,6 @@ golang.org/x/oauth2 v0.0.0-2018082121233
>  golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421/go.mod h1:gOpvHmFTYa4IltrdGE7lF6nIHvwfUNPOp7c8zoXwtLw=
>  golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45/go.mod h1:gOpvHmFTYa4IltrdGE7lF6nIHvwfUNPOp7c8zoXwtLw=
>  golang.org/x/oauth2 v0.0.0-20191202225959-858c2ad4c8b6/go.mod h1:gOpvHmFTYa4IltrdGE7lF6nIHvwfUNPOp7c8zoXwtLw=
> -golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d h1:TzXSXBo42m9gQenoE3b9BGiEpg5IG2JkU5FkPIawgtw=
>  golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d/go.mod h1:gOpvHmFTYa4IltrdGE7lF6nIHvwfUNPOp7c8zoXwtLw=
>  golang.org/x/oauth2 v0.0.0-20200902213428-5d25da1a8d43/go.mod h1:KelEdhl1UZF7XfJ4dDtk6s++YSgaE7mD/BuKKDLBl4A=
>  golang.org/x/oauth2 v0.0.0-20201109201403-9fd604954f58/go.mod h1:KelEdhl1UZF7XfJ4dDtk6s++YSgaE7mD/BuKKDLBl4A=
> Vendor Verify failed.
> ```

Fixed. Thanks

**Comment by liggitt** [comment]
sweeping changes to init blocks, noticed this:
```diff
 func init() {
-	defaultWorker = newWorker()
+	defaultWorker = NewMeter().(*worker)
 	go defaultWorker.start()
 	internal.DefaultRecorder = record
 }
```

looks like the opencensus dependency starts a background goroutine in an init block, which we don't want simply because it is linked into a build. Can we spawn an issue to drive removal of that implicit background behavior?

**Comment by liggitt** [comment]
```
< 	"totalDependencies": 390,
< 	"maxDepthOfDependencies": 33,
< 	"transitiveDependencies": 263
---
> 	"totalDependencies": 388,
> 	"maxDepthOfDependencies": 29,
> 	"transitiveDependencies": 262
```
at least that's moving in the right direction

**Comment by pacoxu** [comment]
> @basantsa1989: Adding label `do-not-merge/contains-merge-commits` because PR contains merge commits, which are not allowed in this repository.
> Use `git rebase` to reapply your commits on top of the target branch. Detailed instructions for doing so can be found [here](https://git.k8s.io/community/contributors/guide/github-workflow.md#4-keep-your-branch-in-sync).
> 
> Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md). If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.

Need update according above to rebase 'merge-commits'.

**Comment by BenTheElder** [comment]
@liggitt re: https://github.com/kubernetes/kubernetes/pull/102926/files#diff-4d8efa3ac884153a7c4647d94411200c951212641b2bdbb3d6408869bf604fe7R33 did we file a thread for this?

**Comment by BenTheElder** [comment]
actually https://github.com/census-instrumentation/opencensus-go/issues/1074 appears to just that, sadly not a new issue, seems no resposne there.

**Comment by pacoxu** [comment]
We upgrade `go.opencensus.io` from v0.22.3 to v0.23.0. It should have problems with both the old and new versions. 
https://github.com/census-instrumentation/opencensus-go/blob/v0.22.3/stats/view/worker.go#L32
https://github.com/census-instrumentation/opencensus-go/blob/v0.23.0/stats/view/worker.go#L32-L36

Should we add it to unwanted dependencies list like #102309 or something like "critical-issue-dependencies-list"? 😓

**Comment by liggitt** [comment]
> Should we add it to unwanted dependencies list like #102309 or something like "critical-issue-dependencies-list"? 😓

not sure that makes it rise to the level of an unwanted dependency, but if you can open an issue to track it (linking to the comment, the line, and the upstream issue), that would be helpful (can add it to https://github.com/orgs/kubernetes/projects/27 as well)

**Comment by MadhavJivrajani** [comment]
/remove-kind design
/kind feature

`kind/design` will soon be removed from `k/k` in favor of `kind/feature`. Relevant discussion can be found here: https://github.com/kubernetes/community/issues/5641

**Comment by dims** [comment]
@basantsa1989 can you please rebase?

**Comment by pacoxu** [comment]
> @basantsa1989 can you please rebase?

@dims I merged this PR as part of #102913 .

**Comment by dims** [comment]
@basantsa1989 @pacoxu so looks like we don't need this anymore? please reopen if needed

**Comment by basantsa1989** [comment]
> @basantsa1989 @pacoxu so looks like we don't need this anymore? please reopen if needed

Yes @dims we are handling this change as part of https://github.com/kubernetes/kubernetes/pull/102913

I was planning to close this once the other PR is merged. I think we can always reopen if needed.

Thanks!

**Pull request closed without merging**